src/plugins/auto/executable_network.cpp

```diff
@@ -676,18 +758,14 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetMetric(const std::st
                 lock.unlock();
                 if (deviceInfo.deviceName.find("GPU") != std::string::npos) {
                     const auto& mode = deviceInfo.config.find(CONFIG_KEY(PERFORMANCE_HINT));
-                    if (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT)) {
-                         std::map<std::string, InferenceEngine::Parameter> options;
-                         options["MODEL_PTR"] = _network.getFunction(); // CNNntework
-                         try {
-                             auto optimalBatchSize = _core->GetMetric(deviceInfo.deviceName,
-                                     METRIC_KEY(OPTIMAL_BATCH_SIZE), options).as<unsigned int>();
-                             auto rangeOfStreams = _core->GetMetric(deviceInfo.deviceName,
-                                     METRIC_KEY(RANGE_FOR_STREAMS), options).as<std::tuple<unsigned int, unsigned int>>();
-                             real = (std::max)(real, std::get<1>(rangeOfStreams) * optimalBatchSize);
-                         } catch (const InferenceEngine::Exception &iie) {
-                             LOG_WARNING("[AUTOPLUGIN]get optimal infer requset num for GPU auto-batch failed :%s", iie.what());
-                         }
+                    // for benchmark through AUTO:CPU,GPU
+                    // SetConfig directly set to CPU/GPU in this case
+                    auto bThroughputEnabledInPlugin =
+                        _core->GetConfig(deviceInfo.deviceName, CONFIG_KEY(PERFORMANCE_HINT)).as<std::string>() == CONFIG_VALUE(THROUGHPUT);
+                    if (bThroughputEnabledInPlugin ||
+                        (mode != deviceInfo.config.end() && mode->second == CONFIG_VALUE(THROUGHPUT))) {
+                        SetOptimalBatchNum(deviceInfo.deviceName);
+                        real = (std::max)(real, _optimalBatchingRequestNum);
```

Comment:
the user created # should be larger than batching size(not sure how to define the muliplex in auto?), otherwise, timeout will be easily triggered in auto batching

Reply:
yes, this is limitation (or rather a "pre-requirement") to get a perf boost from the auto-batching.
Notice that alternative is to communicate the _app's_ limitations to the OV instead (so it adapts the batch size accordingly), via HINT_NUM_REQUESTS- in the cases when the #requests that the app will actually needs is known in advance

Reply:
Hi maxim,
1)  I'll add this key, a question: 
for the PERFORMANCE_HINT_NUM_REQUESTS, the original usage in ie_core is like this, std::min(requests, optimalBatchSize), the minimum one is used:
https://github.com/openvinotoolkit/openvino/pull/7883/commits/d276f6bf0e360dc53ca867918286c08a7833c18f#diff-f4dddf68fff38766cf43b5c758e47e1e1f9e3d395dc5cea2048c662935fc82d3L619
   while in batch plugin,  the max of (hint num request, batchsize) is used, I will follow this one in auto, please share if other concern
2) about the mismatching of request# in auto (batchsize * range streams) and batch plugin (batchsize * optimal number request),  will batching switch to use nstreams also?

Reply:
>  1. while in batch plugin,  the max of (hint num request, batchsize) is used,

Which code exactly do you mean?

> 2. about the mismatching of request# in auto (batchsize * range streams)

streams are somewhat better, but may miss all those (multiply by 2) tweaking on the GPU plugin side.... we will need to close on this by the DG2

Reply:
> > 1. while in batch plugin,  the max of (hint num request, batchsize) is used,
> 
> Which code exactly do you mean?
> 
in batch plugin:
https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/auto_batch/auto_batch.cpp#L440
in the last commit of #7883 :
https://github.com/openvinotoolkit/openvino/pull/7883/commits/d276f6bf0e360dc53ca867918286c08a7833c18f#diff-f4dddf68fff38766cf43b5c758e47e1e1f9e3d395dc5cea2048c662935fc82d3L619
looks like the perfhint is used for deciding on  difference purpose? (batchsize in the reverted logic in ie core) vs (optimal infer request in batch plugin)

Reply:
I see, the code that does the std::max(HINTS_REQUESTS, optimal_batch_size) is rather handling the opposite case (when the batch size is less than the #requsts that user wants to create)... in this case we woudn't return a value smaller than user needs anyway. 
the  primary case (optimal_batch_size > #requets) avoids permanent timeout by handling this at the batch selection stage, tis code is now going to be back : https://github.com/openvinotoolkit/openvino/pull/9724/files#diff-d004c7047eb69283807e51b6ff1ef834b76a5b44339b2679d8caa9315d07cdc2R746